packages/orm/src/getProperties.ts

```diff
@@ -92,9 +92,20 @@ export function getProperties(meta: EntityMetadata): Record<string, any> {
     ),
   );
 
+  // Since our fake instance is actually generating the callbacks for our lazy fields, it will be captured in any
+  // lambdas created.  If any of them reference `this`, then they'll actually be referencing the fake instance.  So we
+  // need to clear out any properties directly on the fake instance now that we're done with it and use a proxy to
+  // intercept any attempts to access `this` from within the callbacks and fail.
+  Object.setPrototypeOf(instance, instancePrototypeProxy);
+  for (const prop of Object.getOwnPropertyNames(instance)) {
+    if (prop !== "__data") delete instance[prop];
+  }
+
   return cached;
 }
 
+const instancePrototypeProxy = new Proxy({}, { get: () => fail("Cannot use 'this' in a property callback") });
```

Comment:
This is good, would be great to eventually link to a page on joist-orm.io that explains why, maybe like an "Advanced / Lazy Relations" page that explains the approach + the gotchas like this one.

https://github.com/joist-orm/joist-orm/issues/1596